hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -196,8 +204,14 @@ std::vector<hardware_interface::CommandInterface> GenericSystem::export_command_
 
 return_type GenericSystem::read()
 {
-  // do loopback
-  for (size_t i = 0; i < joint_states_.size(); ++i) {
+  // apply offset to positions only
+  for (size_t j = 0; j < joint_states_[0].size(); ++j) {
+    if (!std::isnan(joint_commands_[0][j])) {
+      joint_states_[0][j] = joint_commands_[0][j] + state_following_offset_;
+    }
+  }
+  // do loopback on all other interfaces
+  for (size_t i = 1; i < joint_states_.size(); ++i) {
```

Comment:
```suggestion
  for (size_t i = POSITION_INTERFACE_INDEX+1; i < joint_states_.size(); ++i) {
```
+1 to show "the rest"

Reply:
... or just add a comment